haskell/private/haskell_impl.bzl

```diff
@@ -223,6 +231,10 @@ use the 'haskell_import' rule instead.
                 depset([c.objects_dir]),
             ]),
             compiler_flags = ctx.attr.compiler_flags,
+            # NOTE We can't have profiling and dynamic code at the
+            # same time, see:
+            # https://ghc.haskell.org/trac/ghc/ticket/15394
+            with_shared = False,
```

Comment:
Pedantry on my part but is there a good reason we use the term "shared" here and "dynamic" elsewhere? Appreciate we have the duality of static/dynamic to contend with but perhaps there's an opportunity to add clarity if not.

Reply:
`Shared` = whether a library is built in a way in that it can be linked dynamically
`Dynamic` = whether a library/binary links against `shared` libraries

Important distinction for the code, but the user is not exposed to the fact (there’s just `linkstatic`). I don’t know whether that’s a good thing or not, but upstream kind of dictates this API (at least if we want to share it, which we want).

Reply:
Yeah. I'm not sure why the Bazel devs chose to use the same attribute name for two quite different meanings. Historical reasons probably.